hardnet1/main.tf

```diff
@@ -34,21 +34,54 @@ resource "google_storage_bucket_iam_member" "member" {
 }
 
 
-data "template_file" "cloud_config" {
-  template = "${file("${path.module}/cloud-config.yaml.tmpl")}"
-
-  vars {
-  }
-}
+data "template_cloudinit_config" "node" {
+  count = "${var.node_count}"
 
-data "template_cloudinit_config" "config" {
   gzip          = false
   base64_encode = false
 
+  part {
+    filename     = "set-up-instance.sh"
+    content_type = "text/x-shellscript"
+    content      = "${file("set-up-instance.sh")}"
+  }
+
   part {
     filename     = "cloud-config.txt"
     content_type = "text/cloud-config"
-    content      = "${data.template_file.cloud_config.rendered}"
+    content      = "${file("cloud-config.yaml")}"
+  }
+
+  part {
+    filename     = "environment.docker"
+    content_type = "text/x-shellscript"
+    content      = "${file("node-specific/node${count.index}/environment.docker")}"
+  }
+
+  part {
+    filename     = "node.certificate.pem"
+    content_type = "text/x-shellscript"
+    content      = "${file("node-specific/node${count.index}/node.certificate.pem")}"
+  }
+
+  part {
+    filename     = "node.key.pem"
+    content_type = "text/x-shellscript"
+    content      = "${file("node-specific/node${count.index}/node.key.pem")}"
+  }
```

Comment:
Unless you want to have stable node addresses, you can possibly dispose of dealing with node certificates, because now you can query (randomly generated) node address from status API:
```
% curl node0.devnet.rchain-dev.tk:40403/status
{"address":"rnode://5da6b63b6f0b67fb6737cf1d46369115ab36e1d3@node0.devnet.rchain-dev.tk?protocol=40400&discovery=40404","version":"RChain Node 0.9.6.git2a2799fe (2a2799fe7bf28e4d95ad75322795f4408786cd5a)","peers":4,"nodes":4}
```
That requires to start bootstrap first though.

Reply:
Great hint.  Sounds like a task for the Orchestrator.